frontend/src/components/doctor/DetailDoctorPage.tsx

```diff
@@ -10,133 +10,112 @@ import {
   Stack,
   Button,
   Flex,
+  Spinner,
 } from "@chakra-ui/react";
-import { fetchIndividualDoctorInfo } from "../api/FetchData";
 
-export type DoctorDetail = {
-  id: number;
-  first_name: string;
-  last_name: string;
-  email: string;
-  gender: string;
-  location: string;
-  insurance: string[];
-  days_available: string[];
-  speciality: string;
-  about: string;
-};
+import { Doctor } from "./DoctorControl"; // type import
+
+// export default function DetailDoctorPage({
+//   individualDoctor,
+// }: {
+//   individualDoctor?: Doctor;
+// })
 
 export default function DetailDoctorPage() {
+  let [individualDoctor, setIndividualDoctor] = useState<Doctor>();
+
   const params = useParams();
-  if (!params.id) {
-    return <div>No Doctor found</div>;
-  }
-  const doctorID = parseInt(params.id, 10);
 
-  const state = fetchIndividualDoctorInfo(doctorID);
-  if (!state) {
-    return <div>Doctor not found</div>;
-  }
+  // ! means I know doctor id always exist -- don't give error
+  const userId = parseInt(params.id!, 10);
+
+  // get request
+
+  useEffect(() => {
+    const getIndividualDoctor = async () => {
+      const request = await fetch(`/api/doctor/${userId}`);
+      const result = (await request.json()) as Doctor;
+
+      console.log(result);
+      setIndividualDoctor(result);
+    };
 
-  return (
+    getIndividualDoctor();
+  }, []);
```

Comment:
Couple of notes about this `useEffect` hook:

- There's a bug now because it will only fetch on mount. The doctor id could change but the data in this component will be stale. You'll need to add `userId` to the dependency array. If you can, you should add the hook eslint rule: https://www.npmjs.com/package/eslint-plugin-react-hooks.
- You always need to add a cancel method to data fetching effects because the user could navigate away before the network call finishes and you'll get an error like, "Can't set state on an unmounted component" because this will still try to call `setIndividualDoctor` even if the component is no longer mounted. Have a look at this link for more info: https://beta.reactjs.org/learn/synchronizing-with-effects#fetching-data